### PR TITLE
Template repository branch protections

### PR DESCRIPTION
**Pull request by owlot**
- Use the module to create template repositories
- Some re-organization of the repositories_*.tf files to make it easier to find the 'repos' key to create new repositories
- Started on the README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,35 @@
 # tf-github
-Terraform code to create github resources
+
+Terraform code to manage GitHub repositories.
+Resources currently managed by this repository are:
+ * GitHub repositories
+ * Repository branch protections
+ * Repository opermissions
```
Comment (ServiceDog) [inline README.md:7]: *permissions typo

File: README.md [hunk 2]
```diff
@@ -1,2 +1,35 @@
...
+ * Generic files for multiple repositories
+
+## CICD
+
+To deploy changes from this repository, we've installed a Terraform app to provide authentication for the Terraform actions performed on the GitHub resources.
+
+Two secrets have been created in the repository settings itself, which are needed to run the GitHub Actions:
```
Comment (ServiceDog) [inline README.md:14]: Might want to say "multiple" instead of "two" in case the number increases, there are three in the table

File: README.md [hunk 3]
```diff
@@ -1,2 +1,35 @@
...
+As soon as the PR is merged to `main`, a workflow will run to deploy the changes with Terraform.
+
+# Repository management
+
+All repositories are created through the internal `github-repository` module, allowing for easy management of branch protections, files and webhooks.
+Repositories that require similar settings, are grouped into seperate files. The configurations provided by these `repositories_*.tf` files also enforce naming conventions.
+
+The following repository types are currently pre-defined:
+ * `tf`  -- Repositories containing Terraform code
+ * `lib-<name>-rust` -- Repositories
```
Comment (ServiceDog) [inline README.md:31]: svc-<name>-rust as well

File: README.md [hunk 4]
```diff
@@ -1,2 +1,35 @@
...
+
+## Adding a new repository
+
+TODO
```
Comment (ServiceDog) [inline README.md:35]: If this isn't planned for this push, might replace with :construction: Coming Soon :construction:

**Review by ServiceDog — CHANGES_REQUESTED** [review]
Nice refactoring, just a few comments on the doc

**Comment by owlot** [comment]
@ServiceDog thx for the comments. I've made some improvements to the README.md file now.

**Review by ServiceDog — APPROVED** [review]

**Review by GoodluckH — APPROVED** [review]

**Pull request merged**